packages/backend/src/auth/auth.service.ts

```diff
@@ -0,0 +1,163 @@
+import express from 'express'
+import { Secret, sign } from 'jsonwebtoken'
+import { User } from '../user/models/user'
+import { RefreshToken } from './models/refreshToken'
+import logger from '../config/logger'
+
+const log = logger('AuthService')
+
+const generateJWT = (
+  userId: string
+): { accessToken: string; expiresIn: number } => {
+  return {
+    accessToken: sign(
+      { userId },
+      process.env.JWT_ACCESS_TOKEN_SECRET as Secret,
+      {
+        expiresIn: process.env.JWT_ACCESS_TOKEN_EXPIRATION_TIME
+      }
+    ),
+    expiresIn: Number(process.env.JWT_ACCESS_TOKEN_EXPIRATION_TIME)
+  }
+}
+
+const generateRefreshToken = (
+  userId: string
+): { refreshToken: string; expiresIn: number } => {
+  return {
+    refreshToken: sign(
+      { userId },
+      process.env.JWT_REFRESH_TOKEN_SECRET as Secret,
+      {
+        expiresIn: process.env.JWT_REFRESH_TOKEN_EXPIRATION_TIME
+      }
+    ),
+    expiresIn: Number(process.env.JWT_REFRESH_TOKEN_EXPIRATION_TIME)
+  }
+}
+
+export const signup = async (req: express.Request, res: express.Response) => {
+  try {
+    const existingUser = await User.query()
+      .where('email', req.body.email)
+      .first()
+
+    if (existingUser) {
+      return res.status(409).json({ error: 'User already exists' })
+    }
+    await User.query().insert(req.body)
+
+    return res.status(201).json({ message: 'Success' })
+  } catch (error) {
+    log.error(error)
+    return res.status(500).json({ error: 'Unable to create user' })
+  }
+}
+
+export const login = async (req: express.Request, res: express.Response) => {
+  try {
+    const {
+      body: { email, password }
+    } = req
+
+    const user = await User.query()
+      .findOne({ email })
+      .select('id', 'email', 'password')
+      .throwIfNotFound()
```

Comment:
throw an `Invalid credentials` error in this case as well

Reply:
It would have been caught by the last catch that throws an invalid credentials with 401 either way.

Also, this is outdated, and the new implementation has a more broad error handling aproach.
Please give it your review